pkg/cron/jailsync.go

```diff
@@ -0,0 +1,79 @@
+package cron
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/robfig/cron"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	cronSchedule = "0 0 * * *" // every 24 hours
+	jailPath     = "/opt/jail"
+)
+
+type jailSync struct {
+	clusters v3.ClusterInterface
+}
+
+// StartJailSyncCron for cleaning up the /opt/jail dir
+func StartJailSyncCron(scaledContext *config.ScaledContext) error {
+	ref := &jailSync{
+		clusters: scaledContext.Management.Clusters(""),
+	}
+
+	schedule, err := cron.ParseStandard(cronSchedule)
+	if err != nil {
+		return fmt.Errorf("error parsing auth refresh cron: %v", err)
+	}
+
+	c := cron.New()
+	job := cron.FuncJob(ref.syncJails)
+	c.Schedule(schedule, job)
+	c.Start()
+
+	return nil
+}
+
+// syncJails removes any unneeded jails from old clusters.
+func (j *jailSync) syncJails() {
+	// Get the clusters from the api to ensure we are up to date
+	clusters, err := j.clusters.List(metav1.ListOptions{})
+	if err != nil {
+		logrus.Warnf("Error listing clusters for jail cleanup: %v", err)
+	}
+
+	clusterMap := make(map[string]v3.Cluster)
+	for _, cluster := range clusters.Items {
+		clusterMap[cluster.Name] = cluster
+	}
+
+	files, err := ioutil.ReadDir(jailPath)
```

Comment:
would it make sense to move this call to the top of the method? so if the directory lookup fails, we return right away and save on the cluster.list call